[Workplace AI] Add Connector for Amazon S3 List and Get File from Bucket (#253753)
elastic/kibana: docs/reference/connectors-kibana/amazon-s3-action-type.md

---
navigation_title: "Amazon S3"
mapped_pages:
  - https://www.elastic.co/guide/en/kibana/current/amazon-s3-action-type.html
applies_to:

>   stack: preview

Reviewer: ```suggestion
  stack: preview 9.4
```

  serverless: preview
---

# Amazon S3 connector [amazon-s3-action-type]

The Amazon S3 connector enables the listing of buckets, objects within a bucket, and downloading of a specified bucket object.

## Create connectors in {{kib}} [define-amazon-s3-ui]

You can create connectors in **{{stack-manage-app}} > {{connectors-ui}}**. 

### Connector configuration [amazon-s3-connector-configuration]

Amazon S3 connectors have the following configuration properties:

AWS Access Key ID
:   The AWS Access Key ID for the account to authenticate to S3.

AWS Secret Access Key
:   The AWS Secret Access Key for the account to authenticate to S3.

AWS region
:   The AWS region to authenticate to.

## Test connectors [amazon-s3-action-configuration]

> You can test connectors as you're creating or editing the connector in {{kib}}. The test verifies connectivity by trying to access a list of available buckets for your authnticated account.

Reviewer 2: Docs typo “authnticated”

## Get API credentials [amazon-s3-api-credentials]

To use the Amazon S3 connector, you need an AWS Access Key ID, Secret Access Key, and region for the account. Follow these steps to create or view your credentials:

1. Sign in to the [AWS Management Console](https://console.aws.amazon.com/).
2. Click on your account name in the top right corner and select **Security Credentials** from the dropdown menu.
   - **note**: If you do not see the **Security Credentials** link, you may need to talk to your AWS administrator to get access.
3. For better security, AWS recommends using IAM (Identity and Access Management) users instead of root account credentials. If prompted, click **Get Started with IAM Users** to access the IAM Dashboard.
4. In the IAM Dashboard, select **Users** from the left navigation menu.
5. Click on an existing IAM user or create a new user with programmatic access.
6. Navigate to the **Security credentials** tab for the selected user.
7. Under **Access keys**, you can view existing access keys (the Access Key ID is visible, but the Secret Access Key is only shown when first created).

> 8. To create a new access key, click **Create access key** and choose the appropriate use case (e.g., **Application running outside AWS**).

Reviewer: ```suggestion
8. To create a new access key, click **Create access key** and choose the appropriate use case (for example, **Application running outside AWS**).
```

   - Download the `.csv` file or copy both the **Access Key ID** and **Secret Access Key** immediately - the secret key cannot be retrieved later.
9.  For the **AWS region**, select the region where your S3 buckets are located (e.g., `us-east-1`, `eu-west-1`). You can find the region in the AWS console URL or in the S3 bucket details.

:::caution
Store your Secret Access Key securely. It will only be displayed once during creation and cannot be recovered. If lost, you must create a new access key.

> :::

Reviewer: ```suggestion
```
This is already stated at step 8 and removing it avoids 2 successive admonitions
